docs/docs/llms/responses-agent-intro/index.mdx

```diff
@@ -0,0 +1,310 @@
+import { APILink } from "@site/src/components/APILink";
+
+# Tutorial: ResponsesAgent Introduction
+
+## What is a ResponsesAgent?
+`ResponsesAgent` is a subclass of `PythonModel` that provides a framework-agnostic way to create an agent model. Authoring a `ResponsesAgent` provides the following benefits:
+- Ensure your model is compatible with OpenAI Responses API (Read more about it in the [Responses API Introduction](./responses-api-intro))
+- Support for multiple messages and intermediate steps
+- Support for multi-agent scenarios
+- Ensure compatibility with MLflow logging, tracing, and model serving
+
+We recommend `ResponsesAgent` instead of <APILink fn="mlflow.pyfunc.ChatModel">`ChatModel`</APILink> and <APILink fn="mlflow.pyfunc.ChatAgent">`ChatAgent`</APILink>, as it has all the benefits of `ChatAgent` and is compatible with OpenAIs Responses API.
+
+
+## Author a ResponsesAgent
+
+To author a ResponsesAgent, create a subclass of the ResponsesAgent and implement a predict method using any agent authoring framework. You must be using `pydantic>=2` to use ResponsesAgent.
+
+When creating output objects, you should only create `ResponsesAgentResponse` and `ResponsesAgentStreamEvent` pydantic objects. The other classes in `mlflow.types.responses_helpers` are for validation only. Below is an example of a tool calling sequence:
```

Comment:
Feels slightly low level / too detailed for something you see when first reading about the interface?

Reply:
Also maybe worth mentioning the streaming stuff later on - and is it worth maybe having a "Getting started" section with this content + then following up with "Streaming" as a more advanced capability?